phys/module_physics_init.F

```diff
@@ -1472,20 +1472,27 @@ SUBROUTINE phy_init ( id, config_flags, DT, restart, zfull, zhalf,     &
    end if
 #endif
 
-!JJS initialize cloud droplet effective radius
+!  Initialize cloud droplet effective radius for Goddard MP and Radiation
+!  Must be (Goddard LW and SW) AND must be (Goddard 3ice OR 4ice)
 #if ( EM_CORE == 1 )
-    do j=jts,jtf
-       do k=kts,ktf
-          do i=its,itf
-             re_cloud_gsfc(i,k,j) = 0.
-             re_rain_gsfc(i,k,j) = 0.
-             re_snow_gsfc(i,k,j) = 0.
-             re_ice_gsfc(i,k,j) = 0.
-             re_graupel_gsfc(i,k,j) = 0.
-             re_hail_gsfc(i,k,j) = 0.
-          end do
-       end do
-     end do
+    
+   if ( ( ( config_flags%ra_lw_physics .EQ. GODDARDLWSCHEME )   .AND. &
```

Comment:
@weiwangncar @dudhia @powersjg 
Folks,
I think this commit fixes the Goddard memory leak I reported in the WRF Release meeting yesterday. On a nested case that used to fail immediately, this new version of the code now works. I just needed to put an IF test around initializing these six 3d arrays, due to packaging they are only allocated full size when running Goddard MP + Goddard radiation. I just started a big regtest with this. Fingers crossed.